content/courses/advanced-ada/chapters/subprograms.rst

```diff
@@ -3,6 +3,287 @@ Subprograms
 
 .. include:: ../../global.txt
 
+.. _Adv_Ada_Operators:
+
+Operators
+---------
+
+Operators are commonly used for variables of scalar types such as
+:ada:`Integer` and :ada:`Float`. In these cases, they replace *usual* function
+calls. (To be more precise, operators are function calls, but written in a
+different format.) For example, we simply write :ada:`A := A + B + C;` when we
+want to add three integer variables. A hypothetical, non-intuitive version of
+this operation could be :ada:`A := Add (Add (A, B), C);`. In such cases,
+operators allow for expressing function calls in a more intuitive way.
+
+Many primitive operators exist for scalar types. We classify them as follows:
+
++--------------------+--------------------------------------------------------+
+| Category           | Operators                                              |
++====================+========================================================+
+| Logical            | :ada:`and`, :ada:`or`, :ada:`xor`                      |
++--------------------+--------------------------------------------------------+
+| Relational         | :ada:`=`, :ada:`/=`, :ada:`<`, :ada:`<=`, :ada:`>`,    |
+|                    | :ada:`>=`                                              |
++--------------------+--------------------------------------------------------+
+| Unary adding       | :ada:`+`, :ada:`-`                                     |
++--------------------+--------------------------------------------------------+
+| Binary adding      | :ada:`+`, :ada:`-`, :ada:`&`                           |
++--------------------+--------------------------------------------------------+
+| Multiplying        | :ada:`*`, :ada:`/`, :ada:`mod`, :ada:`rem`             |
++--------------------+--------------------------------------------------------+
+| Highest precedence | :ada:`**`, :ada:`abs`, :ada:`not`                      |
++--------------------+--------------------------------------------------------+
+
+.. admonition:: In the Ada Reference Manual
+
+    - `4.5 Operators and Expression Evaluation <http://www.ada-auth.org/standards/12rm/html/RM-4-5.html>`_
+
+User-defined operators
+~~~~~~~~~~~~~~~~~~~~~~
+
+For non-scalar types, not all operators are defined. For example, it wouldn't
+make sense to expect a compiler to include an addition operator for a record
+type with multiple components. An exception to this rule are the
```

Comment:
Maybe not worth changing, but since there are two exceptions to the rule, the sentence should use the plural, such as "Exceptions to this rule are..."